src/main/java/frc/robot/RobotContainer.java

```diff
@@ -47,20 +52,48 @@ public RobotContainer() {
      * {@link edu.wpi.first.wpilibj2.command.button.JoystickButton}.
      */
     private void configureButtonBindings() {
-        Joystick altJoystick = new Joystick(3);
-
-        JoystickButton autoButton = new JoystickButton(altJoystick, 5);
 
-        JoystickButton succButton = new JoystickButton(altJoystick, 1);
-        
+        // Joysticks
+        Joystick leftJoystick = new Joystick(0);
+        Joystick rightJoystick = new Joystick(1);
+        Joystick altJoystick = new Joystick(2);
         
-        double speed = altJoystick.getY();
-        double rotation = altJoystick.getX();
-        double throttle = altJoystick.getThrottle();
+        JoystickButton btnLauncherSolenoid = new JoystickButton(altJoystick, 1);
+        JoystickButton btnAlign = new JoystickButton(leftJoystick, 1);
+        JoystickButton btnIntakeIn = new JoystickButton(altJoystick, 3);
+        JoystickButton btnIntakeOut = new JoystickButton(altJoystick, 4);
+        JoystickButton btnCPExtend = new JoystickButton(leftJoystick, 4); // toggle: use "toggleWhenPressed" method to set command
+        JoystickButton btnWinch = new JoystickButton(altJoystick, 8);
+        JoystickButton btnCPMotor = new JoystickButton(leftJoystick, 3);
+        JoystickButton btnLauncherMotor = new JoystickButton(altJoystick, 12);
+        JoystickButton btnLauncherIdle = new JoystickButton(altJoystick, 10); // toggle: use "toggleWhenPressed" method to set command
+        JoystickButton btnLauncherMotorClose = new JoystickButton(altJoystick, 11);
+        JoystickButton btnLauncherMotorDynamic = new JoystickButton(altJoystick, 9);
+        JoystickButton btnSlowMovement = new JoystickButton(rightJoystick, 1);
+        JoystickButton btnIntakeSolenoid = new JoystickButton(altJoystick, 2); // toggle: use "toggleWhenPressed" method to set command
+        JoystickButton btnScissorExtend = new JoystickButton(altJoystick, 7);
+        JoystickButton btnColorSensor = new JoystickButton(leftJoystick, 5);
+        JoystickButton btnCPStop = new JoystickButton(leftJoystick, 2);
+        JoystickButton btnInvertYAxis = new JoystickButton(leftJoystick, 6);
+        JoystickButton btnRotationSensitivity = new JoystickButton(rightJoystick, 1);
+        JoystickButton btnIntakeBottomOut = new JoystickButton(altJoystick, 6);
+
+        driveSubsystem.setDefaultCommand(new RunCommand(() -> driveSubsystem.drive(-leftJoystick.getY() * 0.7, rightJoystick.getX() * 0.7), driveSubsystem));
+        shooterSubsystem.setDefaultCommand(new RunCommand(() -> {
+            shooterSubsystem.shooter(Math.abs(altJoystick.getY()));
+            System.out.println("Spinning");
+        }, shooterSubsystem));
+
+        btnLauncherSolenoid
+            .whenPressed(new InstantCommand(() -> shooterSubsystem.activatePiston(), shooterSubsystem))
+            .whenInactive(new InstantCommand(() -> shooterSubsystem.lowerPiston(), shooterSubsystem));
```

Comment:
Yeah it's a bit confusing, and it gets even worse when you throw in "whileX." Or the fact that "whenPressed" and "whenActive" are equivalent.

Also, I feel like this "whenActive" turn on "whenInactive" turn off pattern isn't going to scale well... Do you think a better option is to use variables inside the subsystems for state and reset the values to 0 every time the subsystem excutes?

Reply:
Lol scroll down. I had that exact same question. I'm not sure what the 'recommended' way of doing things is. I'm going through the wpilib docs now to see if I can find anything

Reply:
> Subsystems can be associated with “default commands” that will be automatically scheduled when no other command is currently using the subsystem. This is useful for continuous “background” actions such as controlling the robot drive, or keeping an arm held at a setpoint. Similar functionality can be achieved in the subsystem’s periodic() method, which is run once per run of the scheduler; teams should try to be consistent within their codebase about which functionality is achieved through either of these methods. Subsystems are represented in the command-based library by the Subsystem interface

So it looks like they know about both, and tell you to be consistent about it. I think this way is easier to understand for beginners, since you can clearly tell that when you push the button, the shooter goes in, and when you release it, it goes back. Rather than the other way is if you push it, it goes in, but if you don't do anything it goes out. I think it was necessary for robotpy because there wasn't the concept of the required subsystems, so you could have overriding commands that might make a motor 'stutter'. With commandbased, one of them will be preferred (by setting some commands as interruptible)

Reply:
That makes sense. Yeah I think you're right that this method is easier for beginners, and doing it the other way was primarily to avoid that stutter. I just worry that, without default commands or a periodic reset to 0, we might forget to disable things...

Reply:
I guess that's what the sim is for :joy: And careful eyes on the code review. I don't think it'll be too big of a problem. If it proves to be an issue you know how to fix it